File: arrays/firstnlast.py
# Given an array of integers nums sorted in ascending order, find the 
# starting and ending position of a given target value.

# If target is not found in the array, return [-1, -1].

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if target not in nums:
            start = -1
            end = -1
        else:
            start = nums.index(target)
            end = start
            for i in range(len(nums)-1,start,-1):
                if nums[i] == target:
                    end = i
                    break
            

        return [start, end]

File: arrays/firstnlast.js
//leetcode first and last position in array

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let start = 0
    let end = 0
    if(!nums.includes(target)){
        start = -1
        end = -1
    }

    for(let i = 0; i < nums.length; i++){
        if(nums[i] === target){
            start = i
            break
        }
    }
    for(let j = nums.length-1; j >= 0; j--){
        if(nums[j] === target){
            end = j
            break
        }
    }
    

    return [start, end]
};

//testing
//input: [5,7,7,8,8,10], 8
//output: [3,4]